tests/test_runner/test_discover_runner.py

```diff
@@ -36,6 +38,64 @@ def change_loader_patterns(patterns):
         DiscoverRunner.test_loader.testNamePatterns = original_patterns
 
 
+@mock.patch.dict(os.environ, {}, clear=True)  # Isolate from real environment
+class DiscoverRunnerArgumentsTests(SimpleTestCase):
+    def setUp(self):
+        # Python 3.8 on macOS defaults to 'spawn' mode
+        start_method_patcher = mock.patch.object(multiprocessing, "get_start_method", return_value='fork')
+        start_method_patcher.start()
+        self.addCleanup(start_method_patcher.stop)
+
+        # Mock a predictable number of CPU cores
+        cpu_count_patcher = mock.patch.object(multiprocessing, "cpu_count", return_value=12)
+        cpu_count_patcher.start()
+        self.addCleanup(cpu_count_patcher.stop)
+
+    def get_parser(self):
+        parser = argparse.ArgumentParser()
+        DiscoverRunner.add_arguments(parser)
+        return parser
+
+    def test_parallel_default(self):
+        result = self.get_parser().parse_args([])
+        self.assertEqual(result.parallel, 0)
+
+    def test_parallel_flag(self):
+        result = self.get_parser().parse_args(["--parallel"])
+        self.assertEqual(result.parallel, 12)
+
+    def test_parallel_auto(self):
+        result = self.get_parser().parse_args(["--parallel", "auto"])
+        self.assertEqual(result.parallel, 12)
+
+    def test_parallel_count(self):
+        result = self.get_parser().parse_args(["--parallel", "17"])
+        self.assertEqual(result.parallel, 17)
+
+    def test_parallel_bad_string(self):
+        with self.assertRaises(SystemExit) as cm, captured_stderr():
+            self.get_parser().parse_args(["--parallel", "unaccepted"])
+
+        exc = cm.exception.__context__
+        self.assertIsInstance(exc, argparse.ArgumentError)
+        self.assertEqual(str(exc), "argument --parallel: 'unaccepted' is not an integer or the string 'auto'")
```

Comment:
I think it's enough to check `stderr`, e.g.
```python
        with self.assertRaises(SystemExit), captured_stderr() as stderr:
            self.get_parser().parse_args(['--parallel', 'unaccepted'])
        msg = "argument --parallel: 'unaccepted' is not an integer or the string 'auto"
        self.assertIn(msg, stderr.getvalue())
```

Reply:
Good point!